install/installer/pkg/config/v1/envvars.go

```diff
@@ -0,0 +1,418 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the MIT License. See License-MIT.txt in the project root for license information.
+
+package config
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/gitpod-io/gitpod/common-go/log"
+	"github.com/gitpod-io/gitpod/installer/pkg/config"
+	"github.com/gitpod-io/gitpod/installer/pkg/config/v1/experimental"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/yaml"
+)
+
+// isTruthy performs a case-insensitive match for values parsed as true
+func isTruthy(input string) bool {
+	trueValues := []string{
+		"1",
+		"y",
+		"yes",
+		"on",
+		"true",
+		"t",
+	}
+
+	for _, v := range trueValues {
+		if strings.ToUpper(input) == strings.ToUpper(v) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// ConfigEnvvars this maps the external environment variables to Golang values and forms part of our public contract
+type ConfigEnvvars struct {
+	AdvancedModeEnabled                  bool     `env:"ADVANCED_MODE_ENABLED"`
+	ComponentProxyServiceType            string   `env:"COMPONENT_PROXY_SERVICE_SERVICETYPE"`
+	ConfigPatch                          string   `env:"CONFIG_PATCH"`
+	CustomizationPatch                   string   `env:"CUSTOMIZATION_PATCH"`
+	DBCloudSQLEnabled                    bool     `env:"DB_CLOUDSQL_ENABLED"`
+	DBCloudSQLInstance                   string   `env:"DB_CLOUDSQL_INSTANCE"`
+	DBCloudSQLServiceAccountName         string   `env:"DB_CLOUDSQL_SERVICE_ACCOUNT_NAME"`
+	DBExternalCertificateName            string   `env:"DB_EXTERNAL_CERTIFICATE_NAME"`
+	DBInClusterEnabled                   bool     `env:"DB_INCLUSTER_ENABLED" envDefault:"1"`
+	Domain                               string   `env:"DOMAIN"`
+	Distribution                         string   `env:"DISTRIBUTION"`
+	ExternalDockerConfig                 string   `env:"EXTERNAL_DOCKER_CONFIG_JSON"`
+	HTTPProxyName                        string   `env:"HTTP_PROXY_NAME"`
+	ImagePullSecretName                  string   `env:"IMAGE_PULL_SECRET_NAME"`
+	LicenseName                          string   `env:"LICENSE_NAME"`
+	LocalRegistryAddress                 string   `env:"LOCAL_REGISTRY_ADDRESS"`
+	LocalRegistryEnabled                 bool     `env:"HAS_LOCAL_REGISTRY"`
+	LocalRegistryHost                    string   `env:"LOCAL_REGISTRY_HOST"`
+	LocalRegistryImagePullConfig         string   `env:"LOCAL_REGISTRY_IMAGE_PULL_DOCKER_CONFIG_JSON"`
+	OpenVSXUrl                           string   `env:"OPEN_VSX_URL"`
+	RegistryDockerConfigEnabled          bool     `env:"REG_DOCKER_CONFIG_ENABLED"`
+	RegistryDockerConfig                 string   `env:"REG_DOCKER_CONFIG_JSON"`
+	RegistryInClusterEnabled             bool     `env:"REG_INCLUSTER_ENABLED" envDefault:"1"`
+	RegistryInClusterStorageType         string   `env:"REG_INCLUSTER_STORAGE"`
+	RegistryInClusterStorageS3BucketName string   `env:"REG_INCLUSTER_STORAGE_S3_BUCKETNAME"`
```

Comment:
Because the `env`'s are also a bit public facing, I think we should have the same consistency here i.e a `_` btw bucket and name.